ext/objspace/objspace_dump.c

```diff
@@ -37,22 +40,104 @@ struct dump_config {
     unsigned int full_heap: 1;
 };
 
-PRINTF_ARGS(static void dump_append(struct dump_config *, const char *, ...), 2, 3);
 static void
-dump_append(struct dump_config *dc, const char *format, ...)
+dump_flush(struct dump_config *dc)
 {
-    va_list vl;
+    if (dc->buffer_len) {
+        if (dc->stream) {
+            fwrite(dc->buffer, 1, dc->buffer_len, dc->stream);
```

Comment:
What if this `fwrite()` writes fewer bytes than `dc->buff_len`?  For instance signals can interrupt writes to pipes, sockets, etc.

Reply:
Good point. I think the previous code wasn't taking any care for that, but I should.